File: src/nodes/if.zig
const nd = @import("node.zig");

pub const NodeIf = struct {
    condition: *nd.Node,
    body: *nd.Node,

    pub fn init(condition: *nd.Node, body: *nd.Node) NodeIf {
        return NodeIf{ .condition = condition, .body = body };
    }

    pub fn get_condition(self: *const NodeIf) *nd.Node {
        return self.condition;
    }

    pub fn get_body(self: *const NodeIf) *nd.Node {
        return self.body;
    }
};

File: src/nodes/_if.rs
use crate::nodes::node::Node;

pub struct NodeIf {
    condition: Node,
    body: Node,
}

impl NodeIf {
    pub fn init(condition: Node, body: Node) -> NodeIf {
        NodeIf { condition, body }
    }
    
    pub fn get_condition(&self) -> &Node {
        &self.condition
    }
    
    pub fn get_body(&self) -> &Node {
        &self.body
    }
}